server/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecMergeIT.java

```diff
@@ -0,0 +1,172 @@
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.engine.Segment;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.Matchers.is;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_METADATA;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_READ;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_WRITE;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_READ_ONLY;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoFailures;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class MultiCodecMergeIT extends OpenSearchIntegTestCase {
+
+    public void testForceMergeMultipleCodecs() throws ExecutionException, InterruptedException {
+
+        Map<String, String> codecMap = Map.of(
+            "best_compression",
+            "BEST_COMPRESSION",
+            "zstd_no_dict",
+            "ZSTD_NO_DICT",
+            "zstd",
+            "ZSTD",
+            "default",
+            "BEST_SPEED"
+        );
+
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            forceMergeMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
+        }
+
+    }
+
+    private void forceMergeMultipleCodecs(String finalCodec, String finalCodecMode, Map<String, String> codecMap) throws ExecutionException,
+        InterruptedException {
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String index = "test-index" + finalCodec;
+
+        // creating index
+        createIndex(
+            index,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put("index.codec", "default")
+                .build()
+        );
+        ensureGreen(index);
+        Set<String> codecSet = new HashSet<>();
+
+        // ingesting and asserting segment codec mode for all four codecs
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            ingestUsingCodec(index, codec.getKey(), codecSet);
+        }
+
+        assertTrue(codecSet.containsAll(codecMap.values()));
```

Comment:
Can it replaced with something like below:

```
assertTrue(getSegments().stream().map(s -> s.getAttributes(index).get("Lucene90StoredFieldsFormat.mode")).collect(Collectors.toSet()).containsAll(codecMap.values()));
```

Reply:
ack, done